akka-stream/src/main/scala/akka/stream/impl/StreamLayout.scala

```diff
@@ -293,6 +293,51 @@ private[akka] object StreamLayout {
   }
 }
 
+private[stream] object V {
+  sealed trait Termination
+  case object Allowed extends Termination
+  case object Completed extends Termination
+  case class Failed(ex: Throwable) extends Termination
+}
+
+private[stream] final class V[T] extends Processor[T, T] {
+  import V._
+  import ReactiveStreamsCompliance._
+
+  private val subscriptionStatus = new AtomicReference[AnyRef]
+  private val terminationStatus = new AtomicReference[Termination]
+
+  override def subscribe(s: Subscriber[_ >: T]): Unit =
+    if (subscriptionStatus.compareAndSet(null, s)) () // wait for onSubscribe
+    else {
+      val sub = subscriptionStatus.getAndSet(s).asInstanceOf[Subscription]
```

Comment:
It is not a spec violation, just that this VirtualProcessor does not support fanout. This is btw one of the drawbacks of this approach, if you use Sink.publisher to access the features a possibly more powerful Publisher to the left, now it is reduced to single-subscription. I think it is fine, just for the record.